### PR TITLE
feat(cli): add --format json for structured machine-readable output

### LINKED ISSUE #151: feat: add --format json structured output for agent consumption
## Summary

Add a `--format json` flag to `docvet check` and all subcommands, producing structured JSON output that AI agents can parse programmatically.

## Motivation

Agents parse JSON natively. Terminal-formatted output requires pattern matching and is fragile. Structured output enables agents to:
- Programmatically iterate over findings
- Map findings to specific files and lines
- Understand severity and category without parsing prose
- Integrate docvet into automated fix loops

## Proposed output schema

```json
{
  "findings": [
    {
      "file": "src/foo/bar.py",
      "line": 42,
      "symbol": "parse_config",
      "rule": "missing-raises",
      "message": "Docstring missing Raises section for ValueError, FileNotFoundError",
      "category": "enrichment",
      "severity": "warning"
    }
  ],
  "summary": {
    "total": 3,
    "by_category": {"enrichment": 2, "freshness": 1},
    "files_checked": 5
  }
}
```

## Design considerations

- `--format` flag with values: `text` (default, current behavior), `json`
- JSON output should include all fields needed for an agent to fix the issue
- Exit codes unchanged: 0 = clean, 1 = findings
- Consider SARIF format as a future option for GitHub Advanced Security integration

## Related

- MCP server (will consume this same structured output internally)
- GitHub Action (#TBD) will use JSON output for PR annotations

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Adds `--format json` to all five subcommands, producing a structured JSON object with a `findings` array and `summary` for programmatic consumption by AI agents and CI pipelines. Previously, docvet only offered terminal and markdown output, requiring regex parsing for automation.

- Add `JSON = "json"` to `OutputFormat` enum and `format_json()` to reporting module
- Wire JSON branch through `_output_and_exit()` pipeline (always emits, even with zero findings)
- Derive `severity` from `category` at serialization time, preserving the 6-field Finding contract
- Document JSON schema and usage in CLI reference docs

Test: `uv run pytest` (934 tests)

Closes #151

feat(reporting): add format_json() for structured JSON output

docs(cli): document --format json schema and usage

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [x] Types pass (`uv run ty check`)
- [ ] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
- JSON serialization in `format_json()` — severity derivation from category
- `_output_and_exit()` JSON branch — always emits (unlike terminal/markdown)
- 23 new tests covering format_json unit tests, CLI integration, and edge cases

### Related
- Story: 23-3-json-structured-output-format-json
- Epic 23: Machine-Readable and Cross-Platform

### CODE REVIEW

File: src/docvet/reporting.py [hunk 1]
```diff
@@ -127,6 +140,61 @@ def format_markdown(findings: list[Finding]) -> str:
...
+    derived ``severity`` field (``"high"`` for required, ``"low"`` for
+    recommended). Always returns a valid JSON object, even when there
+    are no findings.
+
+    Args:
+        findings: List of findings to format.
+        file_count: Number of files that were checked.
+
+    Returns:
+        JSON string with ``indent=2`` formatting.
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/docvet/reporting.py]: Keeping as-is — the trailing newline is a consistent convention across all three `format_*` functions (`format_terminal`, `format_markdown`, `format_json`) and none of them document it in their Returns section. Each has a dedicated `test_trailing_newline` test that enforces the contract. Adding it to one docstring but not the others would create an inconsistency.
